src/resolve/sourceComponent.ts

```diff
@@ -218,6 +218,9 @@ export class SourceComponent implements MetadataComponent {
   }
 
   get fullName(): string {
+    if (this.type.ignoreTypeName) {
```

Comment:
NIT: the naming makes this line confusing, if `ignoreTypeName` is true, we return `this.type.name` which I wouldn't expect to happen if I set `ignoreTypeName=true`. From the JSDoc, what if we change `ignoreTypeName` to `ignoreParsedFileName`

But I confirmed the change works, 

- deploy `CustomLabels.labels-meta.xml`
-  deploys `Labels.labels-meta.xml`
- deploys `force-app/main/default/myLabels/Labels.labels-meta.xml`
- retrieves into `myLabels/Labels...`
- retrieves into `labels/CustomLabels..`

Reply:
I test this on MPD deploy-retrieve scenarios and did `source:convert` with it, too.